starcompactor/extractors/machine.py

```diff
@@ -0,0 +1,152 @@
+# coding: utf-8
+from __future__ import absolute_import, division, print_function
+import bz2
+import csv
+import datetime
+import logging
+import re 
+import traceback
+
+from os import listdir
+from os.path import isfile, join
+
+LOG = logging.getLogger(__name__)
+
+BACKUP_FILE_DATE_FORMAT = '%Y%m%d'
+BACKUP_FILE_PREFIX = 'mysql_backup_{date}'
+NOVA_COMPUTE_HOST_BINARY = 'nova-compute'
+
+def get_machine_event(process_no, start_date, end_date, backup_path):
```

Comment:
@diurnalist I tried importing to an actual database from backup sql file, but it would be very slow. Processing one file would take several minutes. Considering we have 4 years of backup files, it would take an extremely long time. I tried to run multiple files in parallel but didn't help. Parsing sql file directly speeds things up a lot. 

When I parse the sql file, I record the column order, so if columns are reordered or new columns are added, it would capture that. 

The backup files are our daily backups on UC storage node.

Reply:
@diurnalist reading text-formatted file is slow as well. I guess the data grows a lot compared to the last time I ran the script. I'll try to optimize it.

Reply:
I suspect it might be slow because it's trying to import the entire database, hm. I wonder if it's possible to tell mysql to only import a few tables from a dump. Something like [this approach](https://tecadmin.net/extract-tables-backup-from-full-mysql-dump/) might work. It involves first splitting the dump coarsely into per-table dumps. That allows you to then import only the tables you need. I think the splitting of the file can probably be done quite quickly, but we'd need to try it out.